redwood-treehouse-lazylayout-view/src/main/kotlin/app/cash/redwood/treehouse/lazylayout/view/ViewLazyColumn.kt

```diff
@@ -77,12 +76,12 @@ internal class ViewLazyColumn<A : AppService>(
   private class LazyContentItemListAdapter<A : AppService>(
     private val treehouseApp: TreehouseApp<A>,
     private val widgetSystem: TreehouseView.WidgetSystem<A>,
-    private val contentHeight: Int,
+    private val placeholder: List<Widget<View>>,
   ) : ListAdapter<LazyContentItem, ViewHolder<A>>(LazyContentItemDiffCallback) {
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder<A> {
-      val container = FrameLayout(parent.context).apply {
-        layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, contentHeight)
-      }
+      val container = FrameLayout(parent.context)
+      // TODO Clone placeholder.
```

Comment:
Cloning View's isn't possible. Unsure how to proceed.

Reply:
Yeah this is an interesting use case. You're basically looking to have a single subset of a composition drive multiple identical trees on the display-side. It won't work for any traditional UI toolkit since even if you could clone them you are disconnecting them from the composition so they wouldn't see updates. That's okay in this specific case since your use is static, but in general we can't assume that.

I need to think about this.

Reply:
I don't see any easy path forward to achieving this, and I think that is partly due to the current design's forced usage of Treehouse for individual rows.

If LazyColumn were solely a Redwood-based widget, then the usage of Treehouse for individual rows would mean we could use normal state-handling mechanisms for this.
```kotlin
TreehouseLazyColumn(
  placeholder: @Composable () -> Unit,
  ..
) {
  LazyColumn(..) {
    items(..) {
      var showPlaceholder by remember { mutableStateOf(false) }
      val callback = remember { object : Callback {
        override fun onCodeLoaded(..) { showPlaceholder = false }
      }}
      if (showPlaceholder) {
        placeholder()
      } else {
        TreehouseContent(..)
      }
    }
  }
}
```
This gives you the composable-lambda-is-a-template functionality but its templating still happens within the Redwood composition–not on the display.

On the display side we would use a `RecyclerView` of `FrameLayout`s and the children of each item would render at each position the same it does today if that conditional was your root view. And equivalent things for other platforms/toolkits.

Non-Treehouse `LazyColumn` is tracked by #605 as it was blocking Emoji Search from working on web. We sorta hacked around it with some inversion of control for now. But it feels like a lazy row/column are fundamentals in the same vein as normal row/column that we aught to provide at the Redwood layer.

Reply:
It'd definitely be preferable if `LazyColumn` were a Redwood-based widget as you said. I don't think I'm fully grasping the implementation though:

1. Would the above snippet be within `TreehouseLazyColumn.kt`? Because if so, idk how we'd access `TreehouseContent` considering it exists within the host.
2. In your proposal, if we have `items(count = 1000) { block(it) }`, would `block(it)` be invoked 1000 times as soon as the `LazyColum` is displayed on screen?

Reply:
`TreehouseContent` is a hypothetical Treehouse-equivalent of `RedwoodContent` which is just something that simplifies hosting Treehouse inside of an existing composition. It's basically what is inside of today's `LazyColumn`. We also might not even need this, which is where your next question is going.

If you did `items(count = 1000)` it would not create all 1000, but instead have a policy which would attempt to start filling the viewport + some amount off-screen (maybe a second viewport's worth). As the widget scrolls, it communicates with the composition to fill the upcoming items. So you could imagine for activity we load something like 30 items eagerly and then the others are deferred until you get within a viewport's height of them.

While I'm confident it could be done, it's certainly complicated and I don't think I have the time to really think about it in depth before leave. I'll try to come up with some workaround for the existing design to accomplish this.